### PR TITLE
feat: Add support for clustered tables

### PR DESCRIPTION
**Pull request by giomerlin**

### CODE REVIEW

**Comment by giomerlin** [comment]
It seems like I can not include anyone as a reviewer so I'm tagging you @davidrabinowitz  directly here, has the process changed in any way?

File: connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala [hunk 1]
```diff
@@ -48,7 +49,7 @@ case class BigQueryWriteHelper(bigQuery: BigQuery,
         s"gs://$bucket/.spark-bigquery-${applicationId}-${UUID.randomUUID()}")
       require(gcsPathOption.isDefined, "Temporary GCS path has not been set")
       gcsPath = new Path(gcsPathOption.get)
-      val fs = gcsPath.getFileSystem(conf)
+      val fs = gcsPath.getFileSystem( conf)
```
Comment (davidrabinowitz) [inline connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala]: nit: Please remove the redundant space

File: connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala [hunk 2]
```diff
@@ -145,7 +152,8 @@ case class BigQueryWriteHelper(bigQuery: BigQuery,
   def saveModeToWriteDisposition(saveMode: SaveMode): JobInfo.WriteDisposition = saveMode match {
     case SaveMode.Append => JobInfo.WriteDisposition.WRITE_APPEND
     case SaveMode.Overwrite => JobInfo.WriteDisposition.WRITE_TRUNCATE
-    case unsupported => throw new UnsupportedOperationException(s"SaveMode $unsupported is currently not supported.")
+    case unsupported => throw
```
Comment (davidrabinowitz) [inline connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala]: nit: please revert redundant change

File: connector/src/main/scala/com/google/cloud/spark/bigquery/SparkBigQueryOptions.scala [hunk 3]
```diff
@@ -164,7 +166,7 @@ object SparkBigQueryOptions {
       filter, schema, maxParallelism, temporaryGcsBucket, intermediateFormat, readDataFormat,
       combinePushedDownFilters, viewsEnabled, materializationProject,
       materializationDataset, partitionField, partitionExpirationMs,
-      partitionRequireFilter, partitionType, createDisposition,
+      partitionRequireFilter, partitionType, clusteredFields,createDisposition,
```
Comment (davidrabinowitz) [inline connector/src/main/scala/com/google/cloud/spark/bigquery/SparkBigQueryOptions.scala]: add space after the comma

File: connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndITSuite.scala [hunk 4]
```diff
@@ -459,6 +464,25 @@ class SparkBigQueryEndToEndITSuite extends FunSuite
...
+      .option("temporaryGcsBucket", temporaryGcsBucket)
+      .option("partitionField", "created_timestamp")
+      .option("clusteredFields", "platform")
+      .mode(SaveMode.Overwrite)
+      .save()
+
+    val tableDefinition = testPartitionedTableDefinition
+    tableDefinition.getTimePartitioning.getField shouldBe "created_timestamp"
+    tableDefinition.getClustering.getFields should contain ("platform")
+  }
```
Comment (davidrabinowitz) [inline connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndITSuite.scala]: Please make sure that the partitioned table is deleted when the test ends

Comment (giomerlin) [reply connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndITSuite.scala]: @davidrabinowitz Thanks for the comments, I fixed all of them, except this one, the table is created inside the test dataset, and this dataset is fully deleted inside the afterAll method, should I explicitly delete the table before deleting the dataset? When I ran the tests, the table didn't exists after the dataset was deleted

**Pull request merged**